model/shell/src/main/java/org/opennms/oce/model/shell/Score.java

```
@@ -0,0 +1,165 @@
+import org.opennms.oce.model.v1.schema.Incidents;
+
+// Derive a score comparing a Set of Tickets to a Base Sample
+@Command(scope = "oce", name = "scoreIncidents", description = "Score Correlated Incidents against a baseline.")
+@Service
+public class Score implements Action {
+
+    private Set<Incident> baseline;
+
+    private Set<Incident> unmatchedBaseline = new HashSet<>();
+
+    private Set<Incident> intersection = new HashSet<>();
+
+    private Set<Incident> sut;
+
+    private Set<Incident> unmatchedSut = new HashSet<>();
+
+    private Set<AlarmRef> baselineAlarms;
+
+    private Set<AlarmRef> sutAlarms;
+
+    private Set<AlarmRef> intersectionAlarms = new HashSet<>();
+
+    private Set<AlarmRef> unmatchedAlarms = new HashSet<>();
+
+    @Argument(index = 0, name = "baseline", description = "This is the path for the baseline incidents.xml file.", required = true, multiValued = false)
+    private String baselineFile;
+
+    @Argument(index = 1, name = "score", description = "This is the path for the incidents.xml to be scored.", required = true, multiValued = false)
+    private String scoreFile;
+
+    @Override
+    public Object execute() throws Exception {
+        // calculate the scores 
+        createSets(Paths.get(baselineFile), Paths.get(scoreFile));
+        // Print scores to SysOut
+        System.out.println("Accuracy: " + getAccuracy());
+        System.out.println("Type I Errors: " + getTypeOneErrorCount());
+        System.out.println("Type II Errors: " + getFalseNegativeCount());
+        //
+        System.out.println("Alarm Accuracy: " + getAlarmAccuracy());
+        return null;
+    }
+
+    public Score() {
+    }
+
+    public Score(Path basepath, Path sutPath) {
+        createSets(basepath, sutPath);
+    }
+
+    private void createSets(Path basepath, Path sutPath) {
+        try {
+            baseline = getIncidents(basepath);
+        } catch (JAXBException | IOException e) {
+            e.printStackTrace();
```

Can we re-throw instead of using an empty set here?